src/api/routes/nlp.router.ts

```
@@ -62,16 +63,16 @@ router.post("/process", async (req: Request, res: Response) => {
 
     // If intent is to add items, search for products
     let products = [];
-    if (result.intent === "add_items" && result.items.length > 0) {
+    if (result.intent === "add_items" && result?.items?.length > 0) {
       try {
         const db = getWalmartDatabaseManager();
-        for (const item of result.items.slice(0, 5)) { // Limit to 5 items
-          const searchResults = await db.walmartProducts.searchProducts(item, 3);
+        for (const item of result?.items?.slice(0, 5)) { // Limit to 5 items
```

Biome (2.1.2)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)